### PR TITLE
add api set_params/get_params in Model

### PR DESCRIPTION
**Pull request by zenghsh3**
- add api set_params/get_params in Network; 
- add unittest of api set_params/get_params;
- add Interface of Network and LayerFunc to solve circular imports; 
- refine parameter_names api in Network, which will ensure order consistent between different instance of same Network;
- run teamcity unittest in both GPU and CPU;

### CODE REVIEW

**Comment by zenghsh3** [comment]
- remove licence in third party code; 
- remove interface of Network and LayerFunc; 
- move get_parameter_pairs and get_parameter_names api from plutils to Network

File: parl/framework/model_base.py [hunk 1]
```diff
@@ -143,6 +143,83 @@ def set_params(self, params, gpu_id):
...
+            else:
+                # for any other type, won't be handled. Eg: set
+                pass
+        return parameter_names
+
+    def _get_parameter_pairs(self, src, target):
+        """ Recursively get pairs of parameter names between src and target
+
+        Args:
+            src: parl.Network/parl.LayerFunc/list/tuple/set/dict
```
Comment (TomorrowIsAnOtherDay) [inline parl/framework/model_base.py]: When writing comments for Args, we usually use the following format:
src(type): explaination about what src is here

File: parl/framework/model_base.py [hunk 2]
```diff
@@ -18,13 +18,13 @@
 import hashlib
 import paddle.fluid as fluid
 from abc import ABCMeta
-from parl.framework.interface import NetworkInterface
+from parl.layers.layer_wrappers import LayerFunc
 from parl.plutils import *
```
Comment (TomorrowIsAnOtherDay) [inline parl/framework/model_base.py:22]: Can we remove this line ? 
It seems that we don't have any function imported from parl.utils currently.

File: parl/framework/model_base.py [hunk 3]
```diff
@@ -143,6 +143,83 @@ def set_params(self, params, gpu_id):
...
             set_value(param_name, param, gpu_id)
 
+    def _get_parameter_names(self, obj):
+        """ Recursively get parameter names in obj,
+            mainly used to get parameter names of a parl.Network.
+            The order of parameter names will be consistent between
+            different instances of same parl.Network.
+
+        Args:
+            obj: parl.Network/parl.LayerFunc/list/tuple/set/dict
```
Comment (TomorrowIsAnOtherDay) [inline parl/framework/model_base.py]: Same comment problems here.

File: parl/framework/model_base.py (hunk 2 above)
Comment (zenghsh3) [reply parl/framework/model_base.py:22]: Here is plutils, we have imported functions from plutils.

File: parl/framework/model_base.py (hunk 1 above)
Comment (zenghsh3) [reply parl/framework/model_base.py]: done

File: parl/framework/model_base.py (hunk 3 above)
Comment (zenghsh3) [reply parl/framework/model_base.py]: done

**Review by TomorrowIsAnOtherDay — APPROVED** [review]

**Pull request merged**